framework/src/org/checkerframework/common/signatureinference/SignatureInferenceScenes.java

```diff
@@ -0,0 +1,726 @@
+            builder.setValue(fieldKey, (List<Object>) obj);
+        } else if (obj instanceof String) {
+            builder.setValue(fieldKey, (String)obj);
+        } else if (obj instanceof Integer) {
+            builder.setValue(fieldKey, (Integer)obj);
+        } else if (obj instanceof Float) {
+            builder.setValue(fieldKey, (Float)obj);
+        } else if (obj instanceof Long) {
+            builder.setValue(fieldKey, (Long)obj);
+        } else if (obj instanceof Boolean) {
+            builder.setValue(fieldKey, (Boolean)obj);
+        } else if (obj instanceof Character) {
+            builder.setValue(fieldKey, (Character)obj);
+        } else if (obj instanceof Class<?>) {
+            builder.setValue(fieldKey, (Class<?>)obj);
+        } else if (obj instanceof Double) {
+            builder.setValue(fieldKey, (Double)obj);
+        } else if (obj instanceof Float) {
+            builder.setValue(fieldKey, (Float)obj);
+        } else if (obj instanceof Enum<?>) {
+            builder.setValue(fieldKey, (Enum<?>)obj);
+        } else if (obj instanceof Enum<?>[]) {
+            builder.setValue(fieldKey, (Enum<?>[])obj);
+        } else if (obj instanceof AnnotationMirror) {
+            builder.setValue(fieldKey, (AnnotationMirror)obj);
+        } else if (obj instanceof Object[]) {
+            builder.setValue(fieldKey, (Object[])obj);
+        } else if (obj instanceof TypeMirror) {
+            builder.setValue(fieldKey, (TypeMirror)obj);
+        } else if (obj instanceof Short) {
+            builder.setValue(fieldKey, (Short)obj);
+        } else if (obj instanceof VariableElement) {
+            builder.setValue(fieldKey, (VariableElement)obj);
+        } else if (obj instanceof VariableElement[]) {
+            builder.setValue(fieldKey, (VariableElement[])obj);
+        } else {
+            ErrorReporter.errorAbort("Unrecognized type: " + obj.getClass());
+        }
+    }
+
+    /**
+     * Returns the ClassSymbol of the class encapsulating
+     * the node n passed as parameter.
+     * <p>
+     * If the receiver of field is an instance of "this", the implementation
+     * obtains the ClassSymbol by using classTree. Otherwise, the ClassSymbol
+     * is from the field's receiver.
+     */
+    // TODO: These methods below could be moved somewhere else.
+    private static ClassSymbol getEnclosingClassSymbol(
+            ClassTree classTree, FieldAccessNode field) {
+        Node receiverNode = field.getReceiver();
+        if (receiverNode instanceof ImplicitThisLiteralNode
+                && classTree != null) {
+            return (ClassSymbol) InternalUtils.symbol(classTree);
+        }
+        TypeMirror type = receiverNode.getType();
+        if (type instanceof ClassType) {
+            TypeSymbol tsym = ((ClassType) type).asElement();
+            return tsym.enclClass();
+        }
+        Tree tree = receiverNode.getTree();
+        Element symbol = InternalUtils.symbol(tree);
```

Comment:
Could you replace the rest of this method with "return getEnclosingClassSymbol(tree)" ?